builtin/providers/azurerm/resource_arm_virtual_machine.go

```diff
@@ -190,7 +208,24 @@ func resourceArmVirtualMachine() *schema.Resource {
 
 						"vhd_uri": {
 							Type:     schema.TypeString,
-							Required: true,
+							Optional: true,
```

Comment:
Ok cool, thanks @tombuildsstuff. I re-added the custom logic, but left the ConflictsWith config in place as well so that it will start working correctly after the other PR is merged (has no effect right now).